tests/lsan.supp

```diff
@@ -10,6 +10,27 @@ leak:qd_policy_c_counts_alloc
 leak:qd_policy_open_fetch_settings
 leak:qdr_error_description
 
+# to be triaged; unit_tests
+leak:http-libwebsockets.c
```

Comment:
Just trying to understand why leak:http-libwebsockets.c was added. Can you please elaborate ?

Reply:
`leak:http-libwebsockets.c` is a narrower version of `leak:*libwebsockets*` which was there previously. (The suppression patterns are matched a substrings, as I mentioned on the Jira) I made the suppression more targeted here.

Reply:
Ok, what you are saying makes sense to me. What I would like to happen is for any leaks in our code in the http-libwebsockets.c to be not ignored. Like so, there is a bunch of code we wrote relating to stats and metrics and if there are any leaks in our code, I would like those to be not suppressed. I do understand that we previously had leak:*libwebsockets*

Reply:
I'd be happy to fix this in a subsequent step, or at least make an attempt at it, but for now, currently there _are_ leaks coming from `http-libwebsockets.c` so the suppression needs to stay here:

https://github.com/jiridanek/qpid-dispatch/runs/1964063810?check_suite_focus=true#step:9:593

```
9: =================================================================
9: ==2292==ERROR: LeakSanitizer: detected memory leaks
9: 
9: Indirect leak of 216 byte(s) in 1 object(s) allocated from:
9:     #0 0x7fa3b6a8bdc6 in calloc (/lib/x86_64-linux-gnu/libasan.so.5+0x10ddc6)
9:     #1 0x7fa3b63ce928 in qd_http_server ../src/http-libwebsockets.c:963
9:     #2 0x7fa3b63b5ec7 in qd_server ../src/server.c:1364
9:     #3 0x7fa3b6207fff in qd_dispatch_prepare ../src/dispatch.c:330
9:     #4 0x7fa3b1208ff4  (/lib/x86_64-linux-gnu/libffi.so.7+0x6ff4)
9:     #5 0x7ffce6d1b37f  ([stack]+0x1f37f)
9: 
9: Indirect leak of 64 byte(s) in 1 object(s) allocated from:
9:     #0 0x7fa3b6a8caa5 in posix_memalign (/lib/x86_64-linux-gnu/libasan.so.5+0x10eaa5)
9:     #1 0x7fa3b6270a94 in sys_cond ../src/posix/threading.c:77
9:     #2 0x7fa3b63ce96d in work_queue_init ../src/http-libwebsockets.c:231
9:     #3 0x7fa3b63ce96d in qd_http_server ../src/http-libwebsockets.c:965
9:     #4 0x7fa3b63b5ec7 in qd_server ../src/server.c:1364
9:     #5 0x7fa3b6207fff in qd_dispatch_prepare ../src/dispatch.c:330
9:     #6 0x7fa3b1208ff4  (/lib/x86_64-linux-gnu/libffi.so.7+0x6ff4)
9:     #7 0x7ffce6d1b37f  ([stack]+0x1f37f)
9: 
9: -----------------------------------------------------
9: Suppressions used:
9:   count      bytes template
9:       1       1160 run_unit_tests.c
9:       1         64 sys_mutex
9:       1         56 qdr_core_subscribe
9:       9       5080 ^_PyObject_Realloc
9:     540     862547 ^PyObject_Malloc$
9:       1         32 ^PyThread_allocate_lock$
9:       4       9897 ^PyMem_Malloc$
9:       1        856 ^_PyObject_GC_Resize$
9:       1        295 ^_PyBytes_Resize$
9:       5    1058064 /libwebsockets.so
9: -----------------------------------------------------
9: 
9: SUMMARY: AddressSanitizer: 280 byte(s) leaked in 2 allocation(s).
 7/37 Test  #9: unit_tests .....................................***Failed    2.03 sec
```

Reply:
(IIRC that leak ^^^ is essentially a test issue. There are subsequent suppressions `leak:^callback_healthz$` and `leak:^callback_metrics$` which looked to me more real, but it's hard to tell. And there is little point in investigating/fixing things when I'm no longer dealing with upstream master branch, but with my own patched version, so any PR I might produce would end up rebased few times down the road...)

Reply:
ok, let's fix this in a subsequent step. Thx

Reply:
https://github.com/apache/qpid-dispatch/pull/1080